packages/medusa/src/services/totals.ts

```diff
@@ -936,6 +936,10 @@ class TotalsService extends TransactionBaseService<TotalsService> {
    * @return the total discounts amount
    */
   getDiscountTotal(cartOrOrder: Cart | Order): number {
+    if (cartOrOrder.items.length === 0) {
```

Comment:
I've added this because if there are no items to discount then return early instead of running the rest of the function.